### DISCUSSION TITLE
serving terrain RGB tiles/DEM COGs

### DISCUSSION THREAD

**Discussion by AndrewAnnex** [developmentseed/titiler#530] [Q&A] [answered]
Possibly a feature request, but I was wondering if programs like dem-tiler (github.com/kylebarron/dem-tiler) have been updated to work with titiler or not, and if not how best to do it (likely just forking dem-tiler and replacing what needs to be replaced with titiler/fastapi equivalents). I didn't find any equivalent projects, and I am not aware if front-ends like deck.gl can handle floating point data directly from COGs served by titler, although if there are that would be great. Otherwise I think I see generally what needs to happen, it's just a question if I got the time to do it.

A number of viewers exist around the cesium 3d-tiles format so that is another thing I am considering as I have a use case that can't assume mercator projection

**Comment by vincentsarago** [comment]
Hi @AndrewAnnex 

> Serving RGB Terrain tile

It should be supported once we do https://github.com/developmentseed/titiler/discussions/493 (next version of titiler, see https://github.com/developmentseed/rio-viz/pull/51/files#diff-19b6c00bbb9931102c6b0f89a64b6c5dc055c3cd339ac3fb0e70419fe8f091e5R111)

>  I am not aware if front-ends like deck.gl can handle floating point data directly from COGs served by titler

I don't think deck.gl can handle other datatype that uint8 and uint16

> the cesium 3d-tiles

[here](https://github.com/developmentseed/titiler-mvt) is an example of writting a custom tiling service to produce something else than images

**Reply by vincentsarago** [reply]
cc @kylebarron

**Reply by AndrewAnnex** [reply]
okay likely the datatype can be worked with as is, will be interesting to test it out. Definitely need to get more familiar with the this stack first though to see what would need to be done or if I could just pre-encode the dem to RGB and serve as-is, although per-tile encoding could open possibilities of encoding data with variable vertical resolution, from my understand how the whole RGB encoding thing works.

**Reply by kylebarron** [reply]
> github.com/kylebarron/dem-tiler

No, I haven't touched it in two years.

> not aware if front-ends like deck.gl can handle floating point data directly from COGs served by titler

Maybe. You'd probably have to call the code to generate a mesh from your data yourself, and then pass that terrain mesh into the deck.gl TerrainLayer. But titiler is easy to extend, so you should be able to have a simple extension that generates a mesh on the server side as well.

**Comment by vincentsarago** [comment]
FYI I've started https://github.com/developmentseed/titiler/pull/542 to add custom algorithms (such as terrarium and terrainrgb)

**Comment by AndrewAnnex** [comment, marked as answer]
@vincentsarago and @kylebarron as you both already saw I've made https://github.com/AndrewAnnex/monticello as a new project to move the functionality of dem-tiler to a titiler extension. I haven't tested it yet beyond the unit tests in the project and I have probably missed things as I am not that experienced with fastapi. But in short the original questions I had are now basically resolved!

**Reply by vincentsarago** [reply]
glad you found your way using Titiler extension and Algorithm classes. I'm always looking for feedback on the documentation so if there are anything missing your struggled with please let us know 🙏 

Anyway, keep cranking on the good work 💪
